src/Sarif.Driver/Sdk/AnalyzeCommandBase.cs

```diff
@@ -657,6 +658,56 @@ private SupportedPlatform GetCurrentRunningOS()
             return context;
         }
 
+        internal static void UpdateLocationsAndMessageWithCurrentUri(IList<Location> locations, Message message, Uri updatedUri)
+        {
+            if (locations == null) { return; }
+
+            foreach (Location location in locations)
+            {
+                ArtifactLocation artifactLocation = location.PhysicalLocation?.ArtifactLocation;
+
+                if (message != null)
+                {
+                    string oldFilePath = artifactLocation.Uri.OriginalString;
+                    string newFilePath = updatedUri.OriginalString;
+
+                    string oldFileName = GetFileNameFromUri(artifactLocation.Uri);
+                    string newFileName = GetFileNameFromUri(updatedUri);
+
+                    if (!string.IsNullOrEmpty(oldFileName) && !string.IsNullOrEmpty(newFileName))
+                    {
+                        for (int i = 0; i < message?.Arguments?.Count; i++)
+                        {
+                            if (message.Arguments[i] == oldFileName)
+                            {
+                                message.Arguments[i] = newFileName;
+                            }
+
+                            if (message.Arguments[i] == oldFilePath)
+                            {
+                                message.Arguments[i] = newFilePath;
+                            }
+                        }
+
+                        if (message.Text != null)
+                        {
+                            message.Text = message.Text.Replace(oldFilePath, newFilePath);
+                            message.Text = message.Text.Replace(oldFileName, newFileName);
+                        }
+                    }
+                }
+
+                artifactLocation.Uri = updatedUri;
+            }
+        }
+
+        internal static string GetFileNameFromUri(Uri uri)
+        {
+            if (uri == null) return null;
```

Comment:
>return null [](start = 29, length = 11)

I am perfectly fine with omitting the braces here (in fact I prefer it). But I thought our convention was to include them.

Reply:
right you are

---
In reply to: [322889973](https://github.com/microsoft/sarif-sdk/pull/1670#discussion_r322889973) [](ancestors = 322889973)